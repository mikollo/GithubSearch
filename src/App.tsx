import React, { Component } from "react";
import Results from "./Results";
import githubApiSearchHelper from "./utils/githubApiSearchHelper";
import { View, TextInput, Alert, Text } from "react-native";
export default class App extends Component<{}> {
  state = {
    data: []
  };

  onType = async (query: string) => {
    try {
      this.setState({ loading: true });
      this.setState({ data: await githubApiSearchHelper(query) }, () => {
        if (this.state.data.length < 1) {
          Alert.alert("There are no matches for Your query");
        }
      });
    } catch (e) {
      console.warn(e);
      Alert.alert(e);
    } finally {
      this.setState({ loading: false });
    }
  };

  render() {
    return (
      <View style={{ padding: 20 }}>
        <TextInput placeholder="Search repos" onChangeText={this.onType} />
        <Results data={this.state.data} />
      </View>
    );
  }
}
