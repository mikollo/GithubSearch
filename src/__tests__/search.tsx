import React from "react";
import renderer from "react-test-renderer";
import App from "../App";
import sampleData from "../utils/sampleGithubResponse";
import mock from "xhr-mock";

beforeEach(() => mock.setup());
afterEach(() => mock.teardown());

test("Renders correctly", async () => {
  const tree = renderer.create(<App />);
  expect(tree.toJSON()).toMatchSnapshot();
});

test("Displays results after user types", async () => {
  // mock  github api
  mock.get("https://api.github.com/search/repositories?q=react", {
    status: 200,
    body: JSON.stringify(sampleData)
  });
  // render app
  const tree = renderer.create(<App />);
  // find TextInput and invoke search
  await tree.root
    .findByProps({ placeholder: "Search repos" })
    .props.onChangeText("react");
  // check snapshot correctness
  // however this generates a little bit too big snapshot
  // it could be improved
  expect(tree.toJSON()).toMatchSnapshot();
});
