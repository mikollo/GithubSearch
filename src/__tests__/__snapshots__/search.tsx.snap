// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Displays results after user types 1`] = `
<View
  style={
    Object {
      "padding": 20,
    }
  }
>
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    placeholder="Search repos"
    underlineColorAndroid="transparent"
  />
  <RCTScrollView
    data={
      Array [
        Object {
          "archive_url": "https://api.github.com/repos/facebook/react/{archive_format}{/ref}",
          "archived": false,
          "assignees_url": "https://api.github.com/repos/facebook/react/assignees{/user}",
          "blobs_url": "https://api.github.com/repos/facebook/react/git/blobs{/sha}",
          "branches_url": "https://api.github.com/repos/facebook/react/branches{/branch}",
          "clone_url": "https://github.com/facebook/react.git",
          "collaborators_url": "https://api.github.com/repos/facebook/react/collaborators{/collaborator}",
          "comments_url": "https://api.github.com/repos/facebook/react/comments{/number}",
          "commits_url": "https://api.github.com/repos/facebook/react/commits{/sha}",
          "compare_url": "https://api.github.com/repos/facebook/react/compare/{base}...{head}",
          "contents_url": "https://api.github.com/repos/facebook/react/contents/{+path}",
          "contributors_url": "https://api.github.com/repos/facebook/react/contributors",
          "created_at": "2013-05-24T16:15:54Z",
          "default_branch": "master",
          "deployments_url": "https://api.github.com/repos/facebook/react/deployments",
          "description": "A declarative, efficient, and flexible JavaScript library for building user interfaces.",
          "downloads_url": "https://api.github.com/repos/facebook/react/downloads",
          "events_url": "https://api.github.com/repos/facebook/react/events",
          "fork": false,
          "forks": 21038,
          "forks_count": 21038,
          "forks_url": "https://api.github.com/repos/facebook/react/forks",
          "full_name": "facebook/react",
          "git_commits_url": "https://api.github.com/repos/facebook/react/git/commits{/sha}",
          "git_refs_url": "https://api.github.com/repos/facebook/react/git/refs{/sha}",
          "git_tags_url": "https://api.github.com/repos/facebook/react/git/tags{/sha}",
          "git_url": "git://github.com/facebook/react.git",
          "has_downloads": true,
          "has_issues": true,
          "has_pages": true,
          "has_projects": true,
          "has_wiki": true,
          "homepage": "https://reactjs.org",
          "hooks_url": "https://api.github.com/repos/facebook/react/hooks",
          "html_url": "https://github.com/facebook/react",
          "id": 10270250,
          "issue_comment_url": "https://api.github.com/repos/facebook/react/issues/comments{/number}",
          "issue_events_url": "https://api.github.com/repos/facebook/react/issues/events{/number}",
          "issues_url": "https://api.github.com/repos/facebook/react/issues{/number}",
          "keys_url": "https://api.github.com/repos/facebook/react/keys{/key_id}",
          "labels_url": "https://api.github.com/repos/facebook/react/labels{/name}",
          "language": "JavaScript",
          "languages_url": "https://api.github.com/repos/facebook/react/languages",
          "license": Object {
            "key": "mit",
            "name": "MIT License",
            "node_id": "MDc6TGljZW5zZTEz",
            "spdx_id": "MIT",
            "url": "https://api.github.com/licenses/mit",
          },
          "merges_url": "https://api.github.com/repos/facebook/react/merges",
          "milestones_url": "https://api.github.com/repos/facebook/react/milestones{/number}",
          "mirror_url": null,
          "name": "react",
          "node_id": "MDEwOlJlcG9zaXRvcnkxMDI3MDI1MA==",
          "notifications_url": "https://api.github.com/repos/facebook/react/notifications{?since,all,participating}",
          "open_issues": 471,
          "open_issues_count": 471,
          "owner": Object {
            "avatar_url": "https://avatars3.githubusercontent.com/u/69631?v=4",
            "events_url": "https://api.github.com/users/facebook/events{/privacy}",
            "followers_url": "https://api.github.com/users/facebook/followers",
            "following_url": "https://api.github.com/users/facebook/following{/other_user}",
            "gists_url": "https://api.github.com/users/facebook/gists{/gist_id}",
            "gravatar_id": "",
            "html_url": "https://github.com/facebook",
            "id": 69631,
            "login": "facebook",
            "node_id": "MDEyOk9yZ2FuaXphdGlvbjY5NjMx",
            "organizations_url": "https://api.github.com/users/facebook/orgs",
            "received_events_url": "https://api.github.com/users/facebook/received_events",
            "repos_url": "https://api.github.com/users/facebook/repos",
            "site_admin": false,
            "starred_url": "https://api.github.com/users/facebook/starred{/owner}{/repo}",
            "subscriptions_url": "https://api.github.com/users/facebook/subscriptions",
            "type": "Organization",
            "url": "https://api.github.com/users/facebook",
          },
          "private": false,
          "pulls_url": "https://api.github.com/repos/facebook/react/pulls{/number}",
          "pushed_at": "2018-11-26T15:53:54Z",
          "releases_url": "https://api.github.com/repos/facebook/react/releases{/id}",
          "score": 154.91496,
          "size": 139774,
          "ssh_url": "git@github.com:facebook/react.git",
          "stargazers_count": 116288,
          "stargazers_url": "https://api.github.com/repos/facebook/react/stargazers",
          "statuses_url": "https://api.github.com/repos/facebook/react/statuses/{sha}",
          "subscribers_url": "https://api.github.com/repos/facebook/react/subscribers",
          "subscription_url": "https://api.github.com/repos/facebook/react/subscription",
          "svn_url": "https://github.com/facebook/react",
          "tags_url": "https://api.github.com/repos/facebook/react/tags",
          "teams_url": "https://api.github.com/repos/facebook/react/teams",
          "trees_url": "https://api.github.com/repos/facebook/react/git/trees{/sha}",
          "updated_at": "2018-11-26T16:54:17Z",
          "url": "https://api.github.com/repos/facebook/react",
          "watchers": 116288,
          "watchers_count": 116288,
        },
      ]
    }
    disableVirtualization={false}
    extraData={Array []}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <Text>
            facebook/react
          </Text>
          <Text>
            116288
          </Text>
          <Image
            source={
              Object {
                "uri": "https://avatars3.githubusercontent.com/u/69631?v=4",
              }
            }
            style={
              Object {
                "borderRadius": 10,
                "height": 20,
                "width": 20,
              }
            }
          />
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;

exports[`Displays results after user types 2`] = `
<View
  style={
    Object {
      "padding": 20,
    }
  }
>
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    placeholder="Search repos"
    underlineColorAndroid="transparent"
  />
  <RCTScrollView
    data={
      Array [
        Object {
          "archive_url": "https://api.github.com/repos/facebook/react/{archive_format}{/ref}",
          "archived": false,
          "assignees_url": "https://api.github.com/repos/facebook/react/assignees{/user}",
          "blobs_url": "https://api.github.com/repos/facebook/react/git/blobs{/sha}",
          "branches_url": "https://api.github.com/repos/facebook/react/branches{/branch}",
          "clone_url": "https://github.com/facebook/react.git",
          "collaborators_url": "https://api.github.com/repos/facebook/react/collaborators{/collaborator}",
          "comments_url": "https://api.github.com/repos/facebook/react/comments{/number}",
          "commits_url": "https://api.github.com/repos/facebook/react/commits{/sha}",
          "compare_url": "https://api.github.com/repos/facebook/react/compare/{base}...{head}",
          "contents_url": "https://api.github.com/repos/facebook/react/contents/{+path}",
          "contributors_url": "https://api.github.com/repos/facebook/react/contributors",
          "created_at": "2013-05-24T16:15:54Z",
          "default_branch": "master",
          "deployments_url": "https://api.github.com/repos/facebook/react/deployments",
          "description": "A declarative, efficient, and flexible JavaScript library for building user interfaces.",
          "downloads_url": "https://api.github.com/repos/facebook/react/downloads",
          "events_url": "https://api.github.com/repos/facebook/react/events",
          "fork": false,
          "forks": 21038,
          "forks_count": 21038,
          "forks_url": "https://api.github.com/repos/facebook/react/forks",
          "full_name": "facebook/react",
          "git_commits_url": "https://api.github.com/repos/facebook/react/git/commits{/sha}",
          "git_refs_url": "https://api.github.com/repos/facebook/react/git/refs{/sha}",
          "git_tags_url": "https://api.github.com/repos/facebook/react/git/tags{/sha}",
          "git_url": "git://github.com/facebook/react.git",
          "has_downloads": true,
          "has_issues": true,
          "has_pages": true,
          "has_projects": true,
          "has_wiki": true,
          "homepage": "https://reactjs.org",
          "hooks_url": "https://api.github.com/repos/facebook/react/hooks",
          "html_url": "https://github.com/facebook/react",
          "id": 10270250,
          "issue_comment_url": "https://api.github.com/repos/facebook/react/issues/comments{/number}",
          "issue_events_url": "https://api.github.com/repos/facebook/react/issues/events{/number}",
          "issues_url": "https://api.github.com/repos/facebook/react/issues{/number}",
          "keys_url": "https://api.github.com/repos/facebook/react/keys{/key_id}",
          "labels_url": "https://api.github.com/repos/facebook/react/labels{/name}",
          "language": "JavaScript",
          "languages_url": "https://api.github.com/repos/facebook/react/languages",
          "license": Object {
            "key": "mit",
            "name": "MIT License",
            "node_id": "MDc6TGljZW5zZTEz",
            "spdx_id": "MIT",
            "url": "https://api.github.com/licenses/mit",
          },
          "merges_url": "https://api.github.com/repos/facebook/react/merges",
          "milestones_url": "https://api.github.com/repos/facebook/react/milestones{/number}",
          "mirror_url": null,
          "name": "react",
          "node_id": "MDEwOlJlcG9zaXRvcnkxMDI3MDI1MA==",
          "notifications_url": "https://api.github.com/repos/facebook/react/notifications{?since,all,participating}",
          "open_issues": 471,
          "open_issues_count": 471,
          "owner": Object {
            "avatar_url": "https://avatars3.githubusercontent.com/u/69631?v=4",
            "events_url": "https://api.github.com/users/facebook/events{/privacy}",
            "followers_url": "https://api.github.com/users/facebook/followers",
            "following_url": "https://api.github.com/users/facebook/following{/other_user}",
            "gists_url": "https://api.github.com/users/facebook/gists{/gist_id}",
            "gravatar_id": "",
            "html_url": "https://github.com/facebook",
            "id": 69631,
            "login": "facebook",
            "node_id": "MDEyOk9yZ2FuaXphdGlvbjY5NjMx",
            "organizations_url": "https://api.github.com/users/facebook/orgs",
            "received_events_url": "https://api.github.com/users/facebook/received_events",
            "repos_url": "https://api.github.com/users/facebook/repos",
            "site_admin": false,
            "starred_url": "https://api.github.com/users/facebook/starred{/owner}{/repo}",
            "subscriptions_url": "https://api.github.com/users/facebook/subscriptions",
            "type": "Organization",
            "url": "https://api.github.com/users/facebook",
          },
          "private": false,
          "pulls_url": "https://api.github.com/repos/facebook/react/pulls{/number}",
          "pushed_at": "2018-11-26T15:53:54Z",
          "releases_url": "https://api.github.com/repos/facebook/react/releases{/id}",
          "score": 154.91496,
          "size": 139774,
          "ssh_url": "git@github.com:facebook/react.git",
          "stargazers_count": 116288,
          "stargazers_url": "https://api.github.com/repos/facebook/react/stargazers",
          "statuses_url": "https://api.github.com/repos/facebook/react/statuses/{sha}",
          "subscribers_url": "https://api.github.com/repos/facebook/react/subscribers",
          "subscription_url": "https://api.github.com/repos/facebook/react/subscription",
          "svn_url": "https://github.com/facebook/react",
          "tags_url": "https://api.github.com/repos/facebook/react/tags",
          "teams_url": "https://api.github.com/repos/facebook/react/teams",
          "trees_url": "https://api.github.com/repos/facebook/react/git/trees{/sha}",
          "updated_at": "2018-11-26T16:54:17Z",
          "url": "https://api.github.com/repos/facebook/react",
          "watchers": 116288,
          "watchers_count": 116288,
        },
      ]
    }
    disableVirtualization={false}
    extraData={Array []}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <Text>
            facebook/react
          </Text>
          <Text>
            116288
          </Text>
          <Image
            source={
              Object {
                "uri": "https://avatars3.githubusercontent.com/u/69631?v=4",
              }
            }
            style={
              Object {
                "borderRadius": 10,
                "height": 20,
                "width": 20,
              }
            }
          />
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
